pkg/resources/function_parameters.go

```diff
@@ -0,0 +1,99 @@
+package resources
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var (
+	functionParametersSchema     = make(map[string]*schema.Schema)
+	functionParametersCustomDiff = ParametersCustomDiff(
+		functionParametersProvider,
+		parameter[sdk.FunctionParameter]{sdk.FunctionParameterEnableConsoleOutput, valueTypeBool, sdk.ParameterTypeFunction},
+		parameter[sdk.FunctionParameter]{sdk.FunctionParameterLogLevel, valueTypeString, sdk.ParameterTypeFunction},
+		parameter[sdk.FunctionParameter]{sdk.FunctionParameterMetricLevel, valueTypeString, sdk.ParameterTypeFunction},
+		parameter[sdk.FunctionParameter]{sdk.FunctionParameterTraceLevel, valueTypeString, sdk.ParameterTypeFunction},
+	)
+)
+
+func init() {
+	functionParameterFields := []parameterDef[sdk.FunctionParameter]{
+		// session params
+		{Name: sdk.FunctionParameterEnableConsoleOutput, Type: schema.TypeBool, Description: "Enable stdout/stderr fast path logging for anonyous stored procs. This is a public parameter (similar to LOG_LEVEL)."},
+		{Name: sdk.FunctionParameterLogLevel, Type: schema.TypeString, Description: "LOG_LEVEL to use when filtering events"},
+		{Name: sdk.FunctionParameterMetricLevel, Type: schema.TypeString, ValidateDiag: sdkValidation(sdk.ToMetricLevel), DiffSuppress: NormalizeAndCompare(sdk.ToMetricLevel), Description: "METRIC_LEVEL value to control whether to emit metrics to Event Table"},
```

Comment:
Nit: `METRIC_LEVEL` has a different format than `Trace level` below. Check other places as well.

Reply:
What format you mean? The description? The description is always taken from Snowflake in case of parameters, so this is how these are described in Snowflake.

Reply:
Yes, I meant the description. It looks ugly :/ Resolving, because this is on Snowflake side.